src/Button/Button.component.tsx

```diff
@@ -3,113 +3,619 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { darken } from 'polished';
+import { transparentize } from 'polished';
 // THEME
 import { colors, fonts, sizes } from '../theme';
+import { Colors } from '../Typography/Typography.component';
+
+type Variant = 'primary' | 'outline' | 'minimal';
+
+type ButtonSize = 'xlarge' | 'large' | 'small' | 'xsmall';
```

Comment:
In other places (e.g. icons) we use `xl`, `lg`, `md`, `sm`, `xs`. Should we switch to use those or include them as aliases?

Reply:
Yeah, I went with all the minimal versions in most of my components. The trouble with creating globals—to your point—is that not all elements have all options (i.e. omitting medium). I would like this to be "consistent" in that it uses the abbreviated format, though.

Reply:
In this case, at least, we have the option (if we want) of shifting the names xlarge -> large and large -> medium, so that medium is the default.